sway-core/src/control_flow_analysis/dead_code_analysis.rs

```diff
@@ -1101,6 +1100,9 @@ fn connect_expression(
             }
             Ok(vec![while_loop_exit])
         }
+        // TODO: do we need to create a node here and connect the leaves to it?
```

Comment:
if there's no node, we can't ever report break or continue as dead code. That's _alright_, but the proper thing to do here is to connect the current leaves to the break and continue, and then return an empty vec for leaves as a result. That means that the points leading up to the break/continue are connected, and then the path breaks since everything directly after break/continue is actually dead code.

Reply:
@sezna: You mean like this?

https://github.com/FuelLabs/sway/commit/609f7077890718faa77e4124d05cf69fe8f0accb#diff-8877b4bcb40460d28f3808f38d7efae417662ed07fe5462b224207d08c910130R1104

Note that some of the other cases here which I looked at for guidance seem a bit off. `WhileLoop` for instance only connects `leaves[0]` and ignores any others. `Array` connects it's sub-expressions as if they're all going to be executed in parallel rather than sequentially. I didn't look at any others.

Reply:
> You mean like this? 

yup LGTM